safehttp/response_writer_test.go

```diff
@@ -67,52 +68,68 @@ func TestResponseWriterWriteTwicePanic(t *testing.T) {
 			},
 		},
 		{
-			name: "Call WriteJSON twice",
+			name: "Call NoContent twice",
 			write: func(w *safehttp.ResponseWriter) {
-				obj := struct{ Field string }{Field: "myField"}
-				w.WriteJSON(obj)
-				w.WriteJSON(obj)
+				w.NoContent()
+				w.NoContent()
 			},
 		},
 		{
-			name: "Call Write then WriteJSON",
+			name: "Call WriteError twice",
 			write: func(w *safehttp.ResponseWriter) {
-				obj := struct{ Field string }{Field: "myField"}
-				w.Write(obj)
-				w.WriteJSON(obj)
+				w.WriteError(safehttp.StatusInternalServerError)
+				w.WriteError(safehttp.StatusInternalServerError)
 			},
 		},
 		{
-			name: "Call WriteTemplate twice",
+			name: "Call Redirect twice",
 			write: func(w *safehttp.ResponseWriter) {
-				w.WriteTemplate(template.Must(template.New("name").Parse("<h1>{{ . }}</h1>")), "This is an actual heading, though.")
-				w.WriteTemplate(template.Must(template.New("name").Parse("<h1>{{ . }}</h1>")), "This is an actual heading, though.")
+				ir := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+				w.Redirect(ir, "/asdf", safehttp.StatusMovedPermanently)
+				w.Redirect(ir, "/asdf", safehttp.StatusMovedPermanently)
 			},
 		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			w := safehttp.NewResponseWriter(safehttp.DefaultDispatcher{}, safehttptest.NewTestResponseWriter(&strings.Builder{}), nil)
+			defer func() {
+				if r := recover(); r == nil {
```

Comment:
@empijei are you okay with this pattern? I've seen some discussions on another PR.

I don't have strong opinions here.

Reply:
Thanks for pointing this out. :) Got used to checking whether `recover()` returns `nil` instead of the other way around.